data-prepper-plugins/parquet-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/parquet/DataTypeChecker.java

```diff
@@ -0,0 +1,93 @@
+        if (field.isPrimitive()) {
+            PrimitiveType primitiveType = field.asPrimitiveType();
+            fieldPrimitiveDataType = primitiveType.getPrimitiveTypeName().name();
+            switch (fieldPrimitiveDataType) {
+                case "INT32":
+                    return simpleGroup.getInteger(fieldIndex, 0);
+                case "BOOLEAN":
+                    return simpleGroup.getBoolean(fieldIndex, 0);
+                case "FLOAT":
+                    return simpleGroup.getFloat(fieldIndex, 0);
+                case "DOUBLE":
+                    return simpleGroup.getDouble(fieldIndex, 0);
+                case "INT96":
+                    return simpleGroup.getInt96(fieldIndex, 0);
+                case "INT64":
+                case "LONG":
+                    return simpleGroup.getLong(fieldIndex,0);
+                case "FIXED_LEN_BYTE_ARRAY":
+                    return simpleGroup.getValueToString(fieldIndex, 0);
+                default:
+                    throw new RuntimeException("The Parquet Codec doesn't support this data-type yet.");
+            }
+        }
+        else if (field.getLogicalTypeAnnotation() != null) {
+                if (field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.dateType())) {
+
+                    return simpleGroup.getInteger(fieldIndex, 0);
+                }
+                else if (field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.timestampType(true, LogicalTypeAnnotation.TimeUnit.MICROS))) {
+
+                    return simpleGroup.getLong(fieldIndex, 0);
+
+                }
+                else if (field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(8,true))
+                || field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(16,true))
+                || field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(32,true))
+                || field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(64,true))
+                || field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(8,false))
+                        || field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(16,false))
+                        || field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(32,false))
+                        || field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.intType(64,false))){
+
+                    return simpleGroup.getInteger(fieldIndex,0);
+                }
+                else if (field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.stringType() )) {
+
+                   return simpleGroup.getValueToString(fieldIndex,0);
+                }
+                else if (field.getLogicalTypeAnnotation().equals(LogicalTypeAnnotation.uuidType())){
+
+                    byte[] uuidBytes = simpleGroup.getBinary(fieldIndex,0).getBytes();
+                    return UUID.fromString(new String(uuidBytes));
```

Comment:
Data Prepper doesn't operate on UUID presently. So, I think we should just make this into a `String` and pass the `String` directly.